docs/admin/federation/index.md

```diff
@@ -0,0 +1,184 @@
+For now, we create this secret in the host Kubernetes cluster (that hosts federation control plane). When we start supporting secrets in federation control plane, we will create this secret there.
+Suppose that our kubeconfig for Kubernetes cluster is at `/cluster1/kubeconfig`, we can run the following command to create the secret:
+
+```shell
+$ kubectl create secret generic cluster1 --namespace=federation --from-file=/cluster1/kubeconfig
+```
+
+Note that the file name should be `kubeconfig` since file name determines the name of the key in the secret.
+
+Now that the secret is created, we are ready to register the cluster. The YAML file for cluster will look like:
+
+```yaml
+apiVersion: v1beta1
+kind: Cluster
+metadata:
+  name: cluster1
+spec:
+  serverAddressByClientCIDRs:
+  - clientCIDR: <client-cidr>
+    serverAddress: <apiserver-address>
+  secretRef:
+    name: <secret-name>
+```
+
+We need to insert the appropriate values for `<client-cidr>`, `<apiserver-address>` and `<secret-name>`.
+`<secret-name>` here is name of the secret that we just created.
+serverAddressByClientCIDRs contains the various server addresses that clients
+can use as per their CIDR. We can set the server's public IP address with CIDR
+`"0.0.0.0/0"` which all clients will match. In addition, if we want internal
+clients to use server's clusterIP, we can set that as serverAddress. The client
+CIDR in that case will be a CIDR that only matches IPs of pods running in that
+cluster.
+
+Assuming our YAML file is located at `/cluster1/cluster.yaml`, we can run the following command to register this cluster:
+
+```shell
+$ kubectl create -f /cluster1/cluster.yaml --cluster=federation-cluster
+
+```
+
+By specifying `--cluster=federation-cluster`, we direct the request to federation apiserver.
+we can ensure that the cluster registration was successful by running:
+
+```shell
+$ kubectl get clusters --context=federation-cluster
+NAME       STATUS    VERSION   AGE
+cluster1   Ready               3m
+```
+
+### Updating KubeDNS
+
+Once the cluster is registered with federation, we are all ready to use it.
+But for it to be able to route federation service requests, we need to restart
+KubeDNS and pass it a `--federations` flag which tells it about valid federation DNS hostnames.
+Value of the flag is of the form `--federations=${FEDERATION_NAME}=${DNS_DOMAIN_NAME}`.
+To do this, we can edit the existing kubedns replication controller to include
+this flag in pod template spec and then delete existing pod. Replication controller will
+recreate the pod with updated template.
+To find the name of existing kubedns replication controller you can run
+
+```shell
+$ kubectl get rc --namespace=kube-system
+```
+
+This will list all the replication controllers. Name of the kube-dns rc will
+look like `kube-dns-v18`. You can then edit it by running:
+
+```shell
+$ kubectl edit rc <rc-name> --namespace=kube-system
+```
+Add the --federations flag as args to kube dns container.
```

Comment:
- Add the `--federations` flag.. (backticks)
- s/dns/DNS
- Also, can you take another pass at writing this? Do you mean add this as an arg when you run the container? It's a little unclear where this flag goes, since you don't give an example